vite.config.ts

```diff
@@ -1,9 +1,16 @@
 import { defineConfig } from 'vite'
 import rubyPlugin from 'vite-plugin-ruby'
 import react from '@vitejs/plugin-react'
+import { readFileSync } from 'fs'
 
+// Match latest non-draft at https://github.com/broadinstitute/single_cell_portal_core/releases
+const version = readFileSync('version.txt', { encoding: 'utf8' })
 
 export default defineConfig({
+  'define': {
+    '__SCP_VERSION__': version,
+    '__FRONTEND_SERVICE_WORKER_CACHE__': process.env.VITE_FRONTEND_SERVICE_WORKER_CACHE
```

Comment:
the service worker cache doesn't really have anything to do with Vite specifically, right?  If we went back to webpacker it would still work.  So renaming the variable to just FRONTEND_SERVICE_WORKER_CACHE is probably clearer

Reply:
It'd indeed be clearer, but Vite unfortunately requires custom environment variables passed in via terminal command to be prefixed with `VITE_`.